include/openmc/event.h

```diff
@@ -46,15 +45,21 @@ struct EventQueueItem{
   #endif
   bool operator<(const EventQueueItem& rhs) const
   {
-    //return std::tie(type, material, E) < std::tie(rhs.type, rhs.material, rhs.E);
-    return E < rhs.E;
+    if (material == rhs.material) {
+      return E < rhs.E;
+    } else {
+      return material < rhs.material;
```

Comment:
Should we just be using the commented out code above (that compares two tuples via `std::tie`)?

Reply:
As I recall, std::tie may not be available in device code with all compilers. We could do some testing to see if this is still the case, but even so, it seems best to avoid usage of STL if there is an easy alternative like this.